apps/dashboard/app/models/hpc_module.rb

```diff
@@ -3,23 +3,29 @@
 # HpcModule is a class representing a module you'd see on an HPC system.
 class HpcModule
   class << self
-    def all(cluster)
-      Rails.cache.fetch("modules_#{cluster}", expires_in: 24.hours) do
-        file = "#{Configuration.module_file_dir}/#{cluster}.json"
-        if File.file?(file) && File.readable?(file)
-          begin
-            JSON.parse(File.read(file)).map do |name, spider_output|
-              spider_output.map do |_, mod|
-                HpcModule.new(name, version: mod['Version'], dependencies: mod['parentAA'], hidden: mod['hidden'])
-              end
-            end.flatten.uniq.reject(&:hidden?)
-          rescue StandardError => e
-            Rails.logger.warn("Did not read #{file} correctly because #{e.class}:#{e.message}")
+    def all(cluster = nil)
```

Comment:
Can this be a keyword parameter instead of a positional one? I think it's just a little nicer because then you can call `all` without anything.